modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/RenameProcessorFactoryTests.java

```diff
@@ -20,44 +20,51 @@
 package org.elasticsearch.ingest.common;
 
 import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.ingest.TestTemplateService;
 import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 import static org.hamcrest.CoreMatchers.equalTo;
 
 public class RenameProcessorFactoryTests extends ESTestCase {
 
+    private RenameProcessor.Factory factory;
+
+    @Before
+    public void init() {
+        factory = new RenameProcessor.Factory(TestTemplateService.instance());
+    }
+
     public void testCreate() throws Exception {
-        RenameProcessor.Factory factory = new RenameProcessor.Factory();
         Map<String, Object> config = new HashMap<>();
         config.put("field", "old_field");
         config.put("target_field", "new_field");
         String processorTag = randomAlphaOfLength(10);
         RenameProcessor renameProcessor = factory.create(null, processorTag, config);
         assertThat(renameProcessor.getTag(), equalTo(processorTag));
-        assertThat(renameProcessor.getField(), equalTo("old_field"));
-        assertThat(renameProcessor.getTargetField(), equalTo("new_field"));
+        assertThat(renameProcessor.getField().newInstance(Collections.emptyMap()).execute(), equalTo("old_field"));
+        assertThat(renameProcessor.getTargetField().newInstance(Collections.emptyMap()).execute(), equalTo("new_field"));
         assertThat(renameProcessor.isIgnoreMissing(), equalTo(false));
     }
 
     public void testCreateWithIgnoreMissing() throws Exception {
-        RenameProcessor.Factory factory = new RenameProcessor.Factory();
         Map<String, Object> config = new HashMap<>();
         config.put("field", "old_field");
         config.put("target_field", "new_field");
         config.put("ignore_missing", true);
         String processorTag = randomAlphaOfLength(10);
         RenameProcessor renameProcessor = factory.create(null, processorTag, config);
         assertThat(renameProcessor.getTag(), equalTo(processorTag));
-        assertThat(renameProcessor.getField(), equalTo("old_field"));
-        assertThat(renameProcessor.getTargetField(), equalTo("new_field"));
+        assertThat(renameProcessor.getField().newInstance(Collections.emptyMap()).execute(), equalTo("old_field"));
+        assertThat(renameProcessor.getTargetField().newInstance(Collections.emptyMap()).execute(), equalTo("new_field"));
```

Comment:
Maybe add a test that does a rename with a `{{var}}` snippet and pass down a map with actual key/value in it?

Reply:
@martijnvg I think I can't do that (maybe I'm wrong here?) easily because I don't have the Mustache script engine available in these tests.
With the `org.elasticsearch.ingest.TestTemplateService` all the tests are using `org.elasticsearch.ingest.TestTemplateService.MockTemplateScript.Factory#newInstance` is called which simply ignores the params Map.

Reply:
whoops, I forgot that.